### DISCUSSION TITLE
Automazione per decelerazione imprevista

### DISCUSSION THREAD

**Discussion by jard159** [indomus/forum#885] [HUB: Home Assistant]
salve a tutti, sto cercando di creare una formula/automazione per ricevere un allarme nel momento in cui avvenga una decelerazione improvvisa... una sorta di allarme collisione. mi spiego meglio: HA installato e funzionante per gestire accensione, spegnimento, allarmi di un nastro trasportatore per l'agricoltura. Attualmente sono in grado di rilevare la velocità del nastro in HA  e tutto funziona correttamente. Quello che sto cercando di ottenere è una automazione/allarme che venga innescato se:

in un arco temporale (ad es. da 3 a 0.1 sec) si verifica una variazione di velocità in diminuzione.

Grazie a tutti!!!

**Comment by snakuzzo** [comment]
Ciao. Potresti usare il platform [trend](https://www.home-assistant.io/integrations/trend/).
Da tarare in base alle tue necessità ovviamente

**Reply by snakuzzo** [reply]
dai un'occhiata anche [QUI](https://community.home-assistant.io/t/compare-last-sensor-value-with-new-one/193927/2?u=snakuzzo)

**Reply by jard159** [reply]
grazie mille! sempre TOP!!!
Provo e ti faccio sapere. Grazie mille!

**Comment by jard159** [comment]
ciao, seguendo la guida che mi hai segnalato sono arrivato a costruire qualcosa ma mi sono bloccato nella creazione del `value-template` e dei relativi `sample_duration` e `min_gradient`

supponendo di viaggiare a X km/h (valore dinamico che può variare) se subisco una decelerazione improvvisa compresa tra -3km/h a -10km/h in 2 secondi... allora ALLARME

Mi date una mano??? Grazie mille.

```
    value_template: >-
     {% if states('sensor.speed_nastro') | float > -3 and states('sensor.speed_nastro') | float < -11 %}
       ALLARME
     {% else %}
     {%- endif %}
```


```
binary_sensor:
  - platform: trend
    sensors:
      speed_nastro_decresce:
        entity_id: sensor.speed_nastro
```
